Userland/Services/SpiceAgent/main.cpp

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright (c) 2021, Kyle Pereira <kyle@xylepereira.me>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include "VDAgent.h"
+#include <AK/Format.h>
+#include <LibC/fcntl.h>
+#include <LibC/unistd.h>
+#include <LibIPC/ServerConnection.h>
+
+static constexpr auto SPICE_DEVICE = "/dev/hvc0p1";
+
+int main()
+{
+    Core::EventLoop loop;
+
+    if (pledge("unix rpath wpath stdio sendfd recvfd", nullptr) < 0) {
+        perror("pledge");
+        return 1;
+    }
+
+    int serial_port_fd = open(SPICE_DEVICE, O_RDWR);
+    if (serial_port_fd < 0) {
+        dbgln("Couldn't open spice serial port!");
+        return 1;
+    }
+
+    if (unveil(SPICE_DEVICE, "rw") < 0) {
+        perror("unveil");
+        return 1;
+    }
+    if (unveil("/tmp/portal/clipboard", "rw") < 0) {
+        perror("unveil");
+        return 1;
+    }
```

Comment:
these should be done before the file-system access above, and the unveil should be locked with `nullptr, nullptr`